ToolkitApp/tabs/src/components/sample/ConnectedResources.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Button } from '@fluentui/react-northstar'
+import { ClipboardCopiedToIcon } from '@fluentui/react-icons-northstar'
+import * as microsoftTeams from "@microsoft/teams-js"; 
+import "./ConnectedResources.css";
+import {TEAMS_CHANNEL_ID, CHAT, CHAT_ID} from './TabConstants';
+
+export function processTeamsContext(setResourceList,resourceList,setTitle){
+    //Get the context of where the tab is currently
+    microsoftTeams.getContext(function(context) {
+        
+        //Check if this is a teams channel
+        if(context.channelId && context.channelId.length !== 0){
+            if(isAlreadyPresent(resourceList, context.channelId)){
+                return; 
+            }
+            var channelId = createItemWithCopy(context.channelId);
+            channelId.header = TEAMS_CHANNEL_ID + context.channelId;
+            setTitle(context.teamName + " > " + context.channelName);
+            setResourceList([channelId])
+        }
+        //Check if it is a chat
+        else if(context.chatId && context.chatId.length !== 0){
+            if(isAlreadyPresent(resourceList, context.chatId)){
+                return; 
+            }
+            setTitle(CHAT);
+            var chatId = createItemWithCopy(context.chatId);
+            chatId.header = CHAT_ID + context.chatId;
+            setResourceList([chatId]);
+        }
+        else{
+            setTitle("");
+            setResourceList([]);
+        }
+    });
+}
+
+function isAlreadyPresent(list, id){
```

Comment:
you can use something like
```
resourceList.some((item => {
    item.header?.includes(id);
}));
```

Reply:
c0399e93467c354cf1c334cfcd9fa9c0a6270d56